### PR TITLE
feat(app, shared-data): add i18n formatting function and create titleCase

### PR DESCRIPTION
**Pull request by jerader**
# Overview

Add i18n formatting function so help standardize how we input strings to i18n key values. The different formatting options are `capitalize`, `sentenceCase`, `uppercase`, and `titleCase`

Example on how to use it:

`const { t, i18n } = useTranslation('protocol_details')`
`{i18n.format(t('liquid_name'), 'sentenceCase')}`

# Test Plan

- test out the formatter

# Changelog

- add formatting function to `i18n` config
- create `titleCase` function in shared-data
- add info in the README doc under `copy management`

# Review requests

- test it out, does it work as you would expect? 
- is the information in the README clear?
- does it make sense to add `titleCase` function in the `JS` folder of `shared-data`? 

# Risk assessment

low

### CODE REVIEW

**Review by koji — APPROVED** [review]
lgtm, thank you for doing this!

**Comment by ewagoner** [comment]
Should that example you give maybe go in the `Copy management` section of `app/README.md`?

File: app/package.json [hunk 1]
```diff
@@ -26,6 +26,7 @@
     "@opentrons/shared-data": "link:../shared-data",
     "@thi.ng/paths": "1.6.5",
     "@types/uuid": "^3.4.7",
+    "change-case": "4.1.2",
```
Comment (shlokamin) [inline app/package.json]: do we need both of these? hesitant to bring in two different packages.

sounds like we definitely want something like `titleCase`, but are we practically ever going to use something like `capitalCase`? if so it looks like lodash start case does the same thing.

 `sentenceCase` also looks like it does the same thing as lodash capitalize which we already have

so it seems like all we really to bring in as a dependency is the`title-case` package. also, it looks like the [title-case](https://www.npmjs.com/package/title-case) package points to the same package as `change-case`. 

and [the file itself](https://github.com/blakeembrey/change-case/blob/master/packages/title-case/src/index.ts) is super small! i think we should just copy this file into shared-data and import from there, and then we can avoid bringing in any new dependencies at all

Comment (koji) [reply app/package.json]: As Shlok say, capitalCase and sentenceCase should be handle by `lodash`.
Less packages would make thing easy 😄

Comment (jerader) [reply app/package.json]: Yeah that makes a lot of sense, thanks! I removed the dependencies and added the `titleCase` to `shared-data` 😄

Comment (ewagoner) [reply app/package.json]: I believe we're obligated under the package's MIT license to include the license itself alongside (or as comments within) the file we're using.

```The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.```

https://opensource.org/license/mit/

Comment (jerader) [reply app/package.json]: Thanks for surfacing that @ewagoner! I removed the usage of the dependencies and used the similar `lodash` functions and made a `titleCase` function in `shared-data`. We we shouldn't need to include the license anymore.

**Review by ewagoner — APPROVED** [review]

**Review by ecormany — APPROVED** [review]
Readme info makes sense and seems to be in the logical place. 🚢

**Review by b-cooper — APPROVED** [review]
✍️

**Pull request merged**